src/Controls/src/Core/Shapes/EllipseExtensions.cs

```diff
@@ -0,0 +1,69 @@
+﻿#nullable disable
+using Microsoft.Maui.ApplicationModel;
+
+namespace Microsoft.Maui.Controls.Shapes
+{
+	public static class EllipseExtensions
+	{
+		public static bool TrySetAppTheme(
+			this Ellipse ellipse,
+			string lightResourceKey,
+			string darkResourceKey,
+			BindableProperty bindableProperty,
+			Brush defaultDark,
+			Brush defaultLight,
+			out object outerLight,
+			out object outerDark)
+		{
+			bool anyValueSet = false;
+			if (!Application.Current.TryGetResource(lightResourceKey, out outerLight))
+			{
+				anyValueSet = true;
+				outerLight = defaultLight;
+			}
+			if (!Application.Current.TryGetResource(darkResourceKey, out outerDark))
+			{
+				anyValueSet = true;
+				outerDark = defaultDark;
+			}
+
+			ellipse.SetAppTheme(bindableProperty, outerLight, outerDark);
+			return anyValueSet;
+		}
+
+		public static bool TrySetDynamicThemeColor(
+			this Ellipse ellipse,
+			string resourceKey,
+			BindableProperty bindableProperty,
+			out object outerColor)
+		{
+			if (Application.Current.TryGetResource(resourceKey, out outerColor))
```

Comment:
@StephaneDelcroix is this the best way to retrieve this resource? It seems like we'd retrieve the static resource once the DataTemplate is realized and added to the visual tree that way it can check all the merged dictionaries?

Like, ideally this could use
`radioButton.TryGetResource` right? That way if the resources are specified on the page or the `RadioButton` itself they'll work?